packages/react/src/hooks/useNotifications.ts

```diff
@@ -99,11 +131,46 @@ export const useNotifications = (
     });
   };
 
+  const markNotificationsAsRead = async (notificationIds: string[]) => {
+    await waitFor(() => Boolean(w3iClient));
+    const w3iClientTruthy = w3iClient as Web3InboxClient;
+
+    w3iClientTruthy
+      .markNotificationsAsRead(notificationIds, account, domain)
+      .catch(setError)
+      .then(() => {
+        setData((notifications) =>
+          notifications.map((notification) => ({
+            ...notification,
+            isRead:
+              notification.isRead || notificationIds.includes(notification.id),
+          }))
```

Comment:
Looks like this works, but it's initially confusing what the `||` does. Something like this would be more clear about what's going on:

```suggestion
          notifications.map((notification) => {
            if (notificationIds.includes(notification.id)) {
              return {
                ...notification,
                isRead: return;
              };
            } else {
              return notification;
            }
          }
```